src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -670,65 +716,143 @@ public static void TestStringComparisonEnum()
+                Assert.False(jsonObject.TryGetJsonObjectPropertyValue("object FIRST", stringComparison,
+                    out objectProperty));
+
+                Assert.Equal(0, jsonObject.GetJsonArrayPropertyValue("array first",
+                   stringComparison).Count);
+                Assert.True(jsonObject.TryGetJsonArrayPropertyValue("array first", stringComparison,
+                    out JsonArray arrayProperty));
+                Assert.Equal(0, arrayProperty.Count);
+
+                Assert.Throws<ArgumentException>(() =>
+                    jsonObject.GetJsonArrayPropertyValue("array FIRST", stringComparison));
+                Assert.False(jsonObject.TryGetJsonArrayPropertyValue("array FIRST", stringComparison,
+                    out arrayProperty));
+
+                Assert.Throws<ArgumentNullException>(() =>
+                   jsonObject.Remove(null, stringComparison));
+            }
+
+            void CheckIgnoreCase(JsonObject jsonObject, StringComparison stringComparison)
+            {
+                Assert.Equal(0, jsonObject.GetJsonObjectPropertyValue("OBJECT first",
+                    stringComparison).GetPropertyNames().Count);
+                Assert.True(jsonObject.TryGetJsonObjectPropertyValue("OBJECT first", stringComparison,
+                    out JsonObject objectProperty));
+                Assert.Equal(0, objectProperty.GetPropertyNames().Count);
+
+                Assert.Throws<ArgumentException>(() =>
+                    jsonObject.GetJsonArrayPropertyValue("OBJECT first", stringComparison));
+                Assert.False(jsonObject.TryGetJsonArrayPropertyValue("OBJECT first", stringComparison,
+                    out JsonArray arrayProperty));
+                Assert.False(jsonObject.TryGetJsonArrayPropertyValue("something different", stringComparison,
+                    out arrayProperty));
+
+                Assert.Equal(0, jsonObject.GetJsonArrayPropertyValue("ARRAY first",
+                   stringComparison).Count);
+                Assert.True(jsonObject.TryGetJsonArrayPropertyValue("ARRAY first", stringComparison,
+                    out arrayProperty));
+                Assert.Equal(0, arrayProperty.Count);
+
+                Assert.Throws<ArgumentException>(() =>
+                    jsonObject.GetJsonObjectPropertyValue("ARRAY first", stringComparison));
+                Assert.False(jsonObject.TryGetJsonObjectPropertyValue("ARRAY first", stringComparison,
+                    out objectProperty));
+                Assert.False(jsonObject.TryGetJsonObjectPropertyValue("something different", stringComparison,
+                    out objectProperty));
+
+                Assert.Throws<ArgumentNullException>(() =>
+                   jsonObject.Remove(null, stringComparison));
+            }
+        }
+        
+        [Fact]
+        public static void TestChangingCurrentCulture()
+        {
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
```

Comment:
Looks like this test is failing on linux.   Since you're modifying a static it may be a threading issue.  You might want to make this a remote executor test.  /cc @tarekgh

Reply:
Right, I misinterpreted logs tab information and assumed it should be OK, because I checked it on both architectures locally and in past got unrelated failures.

Reply:
IMO don’t test current culture. It is simplest to test the difference between ordinal and ordinal ignorecase

Reply:
Ok, sure. I can remove it for now and we can consider adding this back in future.